frontend/ui/widget/statuswidget.lua

```diff
@@ -0,0 +1,653 @@
+local InputContainer = require("ui/widget/container/inputcontainer")
+local FrameContainer = require("ui/widget/container/framecontainer")
+local CenterContainer = require("ui/widget/container/centercontainer")
+local RightContainer = require("ui/widget/container/rightcontainer")
+local LeftContainer = require("ui/widget/container/leftcontainer")
+local HorizontalGroup = require("ui/widget/horizontalgroup")
+local VerticalGroup = require("ui/widget/verticalgroup")
+local HorizontalSpan = require("ui/widget/horizontalspan")
+local VerticalSpan = require("ui/widget/verticalspan")
+local InputText = require("ui/widget/inputtext")
+local ToggleSwitch = require("ui/widget/toggleswitch")
+local Button = require("ui/widget/button")
+local ProgressWidget = require("ui/widget/progresswidget")
+local LineWidget = require("ui/widget/linewidget")
+local TextWidget = require("ui/widget/textwidget")
+local ImageWidget = require("ui/widget/imagewidget")
+local CloseButton = require("ui/widget/closebutton")
+local InputDialog = require("ui/widget/inputdialog")
+
+local UIManager = require("ui/uimanager")
+local Geom = require("ui/geometry")
+local Blitbuffer = require("ffi/blitbuffer")
+local Screen = require("device").screen
+local Font = require("ui/font")
+local TimeVal = require("ui/timeval")
+local RenderText = require("ui/rendertext")
+
+local DocSettings = require("docsettings")
+local DEBUG = require("dbg")
+local util = require("util")
+local _ = require("gettext")
+
+--[[
+--Save into sdr folder addtional section
+["summary"] = {
+    ["rating"] = 5,
+    ["note"] = "Some text",
+    ["status"] = "Reading"
+    ["modified"] = "24.01.2016"
+},]]
+local StatusWidget = InputContainer:new {
+    settings = nil,
+    thumbnail = nil,
+    props = nil,
+    star = {},
+    summary = {
+        rating = nil,
+        note = nil,
+        status = "",
+        modified = "",
+    },
+    stats = {
+        total_time_in_sec = 0,
+        performance_in_pages = nil,
+        pages = nil,
+
+    }
+}
+
+function StatusWidget:init()
+    self.stats.pages = self.document:getPageCount()
+    self:getStatisticsSettings()
+    if self.settings then
+        self.summary = self.settings:readSetting("summary")
+    end
+
+    self.small_font_face = Font:getFace("ffont", 15)
+    self.medium_font_face = Font:getFace("ffont", 20)
+    self.large_font_face = Font:getFace("ffont", 25)
+
+    self.star = Button:new {
+        icon = "resources/icons/stats.star.empty.png",
+        bordersize = 0,
+        radius = 0,
+        margin = 0,
+        enabled = true,
+        show_parent = self,
+    }
+
+    local statusContainer = FrameContainer:new {
+        dimen = Screen:getSize(),
+        background = Blitbuffer.COLOR_WHITE,
+        bordersize = 0,
+        padding = 0,
+        self:showStatus(),
```

Comment:
`init` will be called when you declare the widget in readerui. This renders the whole widget every time a user opens a book. Can we make it a lazy loading feature so `showStatus` is only called when user clicks the `Status` item on the menu?